bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PageImpl.java

```diff
@@ -241,7 +244,32 @@ public String getExportedType() {
                                               @Nonnull Class<T> modelClass) {
         Map<String, T> itemWrappers = new LinkedHashMap<>();
 
-        for (final Resource child : slingModelFilter.filterChildResources(request.getResource().getChildren())) {
+        Iterable<Resource> children = null;
+        Template template = null;
+
+        if (this.currentPage != null) {
+            template = this.currentPage.getTemplate();
+        }
+
+        // If the template is editable
+        if (template != null && template.hasStructureSupport()) {
+            ComponentContext componentContext = WCMUtils.getComponentContext(request);
+
+            if (componentContext != null) {
+                TemplateManager templateManager = resolver.adaptTo(TemplateManager.class);
+
+                // Resources backed by the template structure
+                if (templateManager != null) {
+                    children = templateManager.getStructureResources(componentContext);
+                }
+            }
+        }
+
+        if (children == null || !children.iterator().hasNext()) {
```

Comment:
@pfauchere, what's the intention here? It seems like if the template provides structure resources then the page will not expose any more its own components.

Reply:
They will be returned by the `TemplateManager#getStructureResources` function extracting the content of the page using the `componentContext`. And extend that content with the one from the structure of the template.

If no structured resources are found, we fallback to the default